marlowe-playground-client/src/Marlowe/Linter.purs

```diff
@@ -359,18 +359,12 @@ addMoneyToEnvAccount amountToAdd accTerm tokenTerm = over _deposits (Map.alter (
 -- | The aim here is to only traverse the contract once since we are concerned about performance with the linting
 -- FIXME: There is a bug where if you create holes with the same name in different When blocks they are missing from
 -- the final lint result. After debugging it's strange because they seem to exist in intermediate states.
-lint :: List ContractPath -> Semantics.State -> Term Contract -> State
-lint unreachablePaths contractState contract = state
-  where
-  choices = contractState ^. (_choices <<< to Map.keys)
-
-  bindings = contractState ^. (_boundValues <<< to Map.keys)
-
-  deposits = contractState ^. (_accounts <<< to (Map.mapMaybe (Just <<< Just)))
-
-  env = (set _letBindings bindings <<< set _deposits deposits <<< set _choicesMade choices) (emptyEnvironment unreachablePaths)
-
-  state = CMS.execState (lintContract env contract) mempty
+lint :: List ContractPath -> Term Contract -> State
+lint unreachablePaths contract =
+  let
+    env = (emptyEnvironment unreachablePaths)
+  in
+    CMS.execState (lintContract env contract) mempty
```

Comment:
I think it's worth mentioning in the PR that we are removing `Semantics.State` from the linting because we will now not lint a partially simulated contract. Especially for the attention of @palas

Reply:
Comment added